src/transcoding/transcode/audio.c

```diff
@@ -79,6 +79,47 @@ _audio_context_sample_rate(TVHContext *self, AVDictionary **opts)
 }
 
 
+#if LIBAVCODEC_VERSION_MAJOR > 59
+static AVChannelLayout
+_audio_context_channel_layout(TVHContext *self, AVDictionary **opts)
+{
+    const AVChannelLayout *channel_layouts =
+        tvh_codec_profile_audio_get_channel_layouts(self->profile);
+    AVChannelLayout ilayout = self->iavctx->ch_layout;
+    AVChannelLayout olayout;
+    AVChannelLayout altlayout;
+    int tmp = 0, ichannels = 0, i;
+    char obuf[64], abuf[64], ibuf[64];
+
+    if (!tvh_context_get_int_opt(opts, "ch_layout", &tmp) &&
+        !(av_channel_layout_from_mask(&olayout, tmp)) && channel_layouts) {
+        ichannels = ilayout.nb_channels;
+        for (i = 0; (av_channel_layout_from_mask(&olayout, channel_layouts[i].u.mask)); i++) {
+            if (av_channel_layout_compare(&olayout, &ilayout)) {
+                break;
+            }
+            if (olayout.nb_channels <= ichannels) {
+                altlayout = olayout;
+            }
+        }
+    }
+    if (tvhtrace_enabled()) {
+        strcpy(obuf, "none");
+        av_channel_layout_describe(&olayout, obuf, sizeof(obuf));
+        strcpy(abuf, "none");
+        av_channel_layout_describe(&altlayout, abuf, sizeof(abuf));
+        strcpy(ibuf, "none");
+        av_channel_layout_describe(&ilayout, ibuf, sizeof(ibuf));
+        tvh_context_log(self, LOG_TRACE, "audio layout selection: old %s, alt %s, in %s",
+                        obuf, abuf, ibuf);
+    }
+    if (olayout.order == AV_CHANNEL_ORDER_UNSPEC)
```

Comment:
Sonarcloud is unhappy with this, as in case the `if` above is not true, it will read from an uninitialized variable. The whole tvhteace enabled and following code should probably be put into the `if`, and an else branch should return the `AV_CHANNEL_LAYOUT_STEREO ` (with a log message that the default was chosen).

Or, if there is a way to initialize the layout above, that could also be done.